src/catalog/catalog_entry/duck_schema_entry.cpp

```diff
@@ -316,11 +317,15 @@ void DuckSchemaEntry::DropEntry(ClientContext &context, DropInfo &info) {
 	// if this is a index or table with indexes, initialize any unknown index instances
 	LazyLoadIndexes(context, *existing_entry);
 
-	// if there is a foreign key constraint, get that information
 	vector<unique_ptr<AlterForeignKeyInfo>> fk_arrays;
 	if (existing_entry->type == CatalogType::TABLE_ENTRY) {
-		FindForeignKeyInformation(existing_entry->Cast<TableCatalogEntry>(), AlterForeignKeyType::AFT_DELETE,
-		                          fk_arrays);
+		// if we have transaction local insertions for this table - clear them
+		auto &table_entry = existing_entry->Cast<TableCatalogEntry>();
+		auto &local_storage = LocalStorage::Get(DuckTransaction::Get(context, catalog));
+		local_storage.DropTable(table_entry.GetStorage());
```

Comment:
Ah this makes it so that we don't create the problematic WAL entry with the Flush that's done in Commit, that looks correct to me 👍

Reply:
Can we verify that a RENAME of a table does not trigger this?

Reply:
I've expanded the tests - it does not but `CREATE OR REPLACE` follows a different codepath. I've now also solved it in that instance.